### ISSUE TITLE
Support gw

### ISSUE THREAD

**Issue by alexanderkjeldaas** [VSCodeVim/Vim#4523]
**Describe the solution you'd like**
I need `gw{motion}` to work in order to edit plain text, like markdown.

**Describe alternatives you've considered**
There are probably other vim commands that can be used, but `gw` is what I know.

**Comment by alexanderkjeldaas** [comment]
Oh, gw is like fill-paragraph in emacs, it will wrap long lines based on textwidth.

**Comment by max-sixty** [comment]
Does `gq` do what you need?

**Comment by alexanderkjeldaas** [comment]
Yes it seems like gq does the same thing(?)

**Comment by J-Fields** [comment]
http://vimdoc.sourceforge.net/htmldoc/change.html#gw

**Comment by berknam** [comment]
Isn't the current action of `gq` actually more similar to the `gw` than the `gq` though? Except for the cursor position. Because `gq` is supposed to take into account the `'formatexpr'` and `'formatprg'` while `gw` isn't and in our case we don't have any of that, so if we changed the cursor situation this would be more like the `gw`.
Or are we aiming to just get a `gw` that is equal to our `gq` but keeping the cursor? I don't know enough about this commands to know if the `'formatexpr'` and `'formatprg'` is even feasible to implement in VSCodeVim, maybe it's not.

**Comment by pbnj** [comment]
I agree that the current behavior of `gq` is a bit jarring to Vim users. As @berknam said, it behaves much more like `gw` (minus the cursor position).

In vim, I set `formatprg` to the appropriate formatting tool based on filetype.

```vimrc
autocmd FileType json setlocal formatprg="prettier --parser=json --stdin-filepath --"
```

Which allows me to `gq{motion}` (e.g. `gggqG` to go to top of file and format until end of file). This also works for Go, Terraform, SQL ...etc. See my [dotfiles](https://github.com/pbnj/dotfiles/blob/master/vim/.vim/vimrc#L336-L351) for reference.

I would be fine if `gq` worked in VS Code by essentially calling `Format Document` or `Format Selection` under the hood.

**Comment by J-Fields** [comment]
What's the difference between `=`/`equalprg` and `gq`/`formatprg` in terms of intention?

**Comment by pbnj** [comment]
Per the [equalprg](https://vimhelp.org/options.txt.html#%27equalprg%27) and [formatprg](https://vimhelp.org/options.txt.html#%27formatprg%27) docs, the way I understand the difference to be is `=` is only concerned with indentation, while `gq` is concerned with text formatting.

**Comment by J-Fields** [comment]
Hmm so maybe we do the following?

- Change `gq` to do what `=` currently does (`Format Selection`)
- Change `=` to do `Reindent Lines`
- Implement `gw` to do what `gq` currently does (reflow lines), but with correct cursor positioning

Then add `equalprg` and/or `formatprg` if someone complains for some reason.
